src/journal/catalog.c

```diff
@@ -94,25 +94,85 @@ const struct hash_ops catalog_hash_ops = {
         .compare = catalog_compare_func
 };
 
+static bool next_header(const char **s) {
+        const char *e;
+
+        e = strchr(*s, '\n');
+
+        /* Unexpected end */
+        if (!e)
+                return false;
+
+        /* End of headers */
+        if (e == *s)
+                return false;
+
+        *s = e + 1;
+        return true;
+}
+
+static int combine_entries(const char *one, const char *two, char **result) {
+        const char *b1, *b2;
+        size_t l1, l2, n;
+        char *dest, *p;
+
+        /* Find split point of headers to body */
+        b1 = one;
+        while (next_header(&b1));
+        b2 = two;
+        while (next_header(&b2));
+
+        l1 = strlen(one);
+        l2 = strlen(two);
+        dest = malloc(l1 + l2 + 1);
```

Comment:
We still prefer `new()` over plain `malloc()` for type safety of the returned value.